diff --git a/lib/bundle/styles/_code.scss b/lib/bundle/styles/_code.scss
--- a/lib/bundle/styles/_code.scss
+++ b/lib/bundle/styles/_code.scss
@@ -61,10 +61,18 @@
 @mixin code-vars($code-color, $code-bg-color) {
+  $highlight-saturation: 10%;
+  $highlight-lightness: -5%;
+
+  @if whiteness($code-bg-color) < 50% {
+    $highlight-saturation: 5%;
+    $highlight-lightness: 5%;
+  }
+
   --code-c-text: #{$code-color};
   --code-c-bg: #{$code-bg-color};
   --code-c-highlight-bg: #{color.scale(
       $code-bg-color,
-      $saturation: if(sass(whiteness($code-bg-color) < 50%), 5%, 10%),
-      $lightness: if(sass(whiteness($code-bg-color) < 50%), 5%, -5%)
+      $saturation: $highlight-saturation,
+      $lightness: $highlight-lightness
     )};
   --code-c-line-number: #{color.scale($code-color, $alpha: -33%)};
 }
diff --git a/lib/client/components/sidebar/SidebarLinks.js b/lib/client/components/sidebar/SidebarLinks.js
--- a/lib/client/components/sidebar/SidebarLinks.js
+++ b/lib/client/components/sidebar/SidebarLinks.js
@@ -20,13 +20,13 @@ export default defineComponent({
     setup(props) {
         const route = useRoute();
         const routePath = useRoutePath();
-        const openGroupIndex = ref(-1);
+        const getActiveGroupIndex = () => props.config.findIndex((item) => isActiveSidebarItem(route, item));
+        const openGroupIndex = ref(getActiveGroupIndex());
         const toggleGroup = (index) => {
             openGroupIndex.value = index === openGroupIndex.value ? -1 : index;
         };
         watchImmediate(routePath, () => {
-            const index = props.config.findIndex((item) => isActiveSidebarItem(route, item));
-            openGroupIndex.value = index;
-        }, { flush: "post" });
+            openGroupIndex.value = getActiveGroupIndex();
+        });
         return () => h("ul", { class: "vp-sidebar-links" }, props.config.map((config, index) => h("li", "children" in config
             ? h(SidebarGroup, {
                 config,
